server/src/test/java/io/crate/integrationtests/LuceneQueryBuilderIntegrationTest.java

```diff
@@ -343,6 +343,66 @@ public void testWhereNotEqualAnyWithLargeArray() throws Exception {
         assertThat(response.rows()[0][0]).isEqualTo(1L);
     }
 
+    @Test
+    public void testWhereNotEqualAnyWithLargeArrayWithIndexOffForIntegerType() throws Exception {
+        // Test overriding of default value 8192 for indices.query.bool.max_clause_count
+        execute("create table t1 (id integer index off) clustered into 2 shards with (number_of_replicas = 0)");
+        execute("create table t2 (id integer index off) clustered into 2 shards with (number_of_replicas = 0)");
+        ensureYellow();
+
+        int bulkSize = NUMBER_OF_BOOLEAN_CLAUSES;
+        Object[][] bulkArgs = new Object[bulkSize][];
+        for (int i = 0; i < bulkSize; i++) {
+            bulkArgs[i] = new Object[]{i};
+        }
+        execute("insert into t1 (id) values (?)", bulkArgs);
+        execute("insert into t2 (id) values (1)");
+        execute("refresh table t1, t2");
+
+        execute("select count(*) from t2 where id != any(select id from t1)");
+        assertThat(response.rows()[0][0]).isEqualTo(1L);
+    }
+
+    @Test
+    public void testWhereNotEqualAnyWithLargeArrayWithColumnStoreOffForIntegerType() throws Exception {
+        // Test overriding of default value 8192 for indices.query.bool.max_clause_count
+        execute("create table t1 (id integer storage with (columnstore = false)) clustered into 2 shards with (number_of_replicas = 0)");
+        execute("create table t2 (id integer storage with (columnstore = false)) clustered into 2 shards with (number_of_replicas = 0)");
+        ensureYellow();
+
+        int bulkSize = NUMBER_OF_BOOLEAN_CLAUSES;
+        Object[][] bulkArgs = new Object[bulkSize][];
+        for (int i = 0; i < bulkSize; i++) {
+            bulkArgs[i] = new Object[]{i};
+        }
+        execute("insert into t1 (id) values (?)", bulkArgs);
+        execute("insert into t2 (id) values (1)");
+        execute("refresh table t1, t2");
+
+        execute("select count(*) from t2 where id != any(select id from t1)");
+        assertThat(response.rows()[0][0]).isEqualTo(1L);
```

Comment:
This test fails with:

```
java.lang.IllegalStateException: FieldExistsQuery requires that the field indexes doc values, norms or vectors, but field 'id' exists and indexes neither of these data structures

	at __randomizedtesting.SeedInfo.seed([E13E474889DC2B55:F8BFE130C941B4F0]:0)
	at org.apache.lucene.search.FieldExistsQuery.rewrite(FieldExistsQuery.java:170)
	at org.apache.lucene.search.ConstantScoreQuery.rewrite(ConstantScoreQuery.java:44)
	at org.apache.lucene.search.BooleanQuery.rewrite(BooleanQuery.java:288)
	at org.apache.lucene.search.ConstantScoreQuery.rewrite(ConstantScoreQuery.java:44)
	at org.apache.lucene.search.ConstantScoreQuery.rewrite(ConstantScoreQuery.java:44)
	at org.apache.lucene.search.IndexSearcher.rewrite(IndexSearcher.java:799)
	at org.apache.lucene.search.IndexSearcher.rewrite(IndexSearcher.java:813)
	at org.apache.lucene.search.IndexSearcher.search(IndexSearcher.java:686)
	at org.apache.lucene.search.IndexSearcher.count(IndexSearcher.java:460)
	at io.crate.execution.engine.collect.count.InternalCountOperation.syncCount(InternalCountOperation.java:166)
	at io.crate.execution.engine.collect.count.InternalCountOperation.lambda$prepareGetCount$2(InternalCountOperation.java:137)
	at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.lang.Thread.run(Thread.java:1589)
```
should be fixed by 6701a5f076badfb3096480595c0ef7eb5587bf47.

Reply:
A smaller test that triggers the same issue:

```
cr> create table t (a int storage with (columnstore = false));                                                                                    
CREATE OK, 1 row affected  (0.263 sec)
cr> insert into t values (1), (null);                                                                                                             
INSERT OK, 2 rows affected  (0.065 sec)
cr> select * from t where a is null;                                                                                                              
IllegalStateException[FieldExistsQuery requires that the field indexes doc values, norms or vectors, but field 'a' exists and indexes neither of these data structures]
```